src/AzureClient/AzureClient.cs

```diff
@@ -357,19 +364,14 @@ public async Task<ExecutionResult> GetJobResultAsync(IChannel channel, string jo
                 return AzureClientError.JobNotCompleted.ToExecutionResult();
             }
 
-            var stream = new MemoryStream();
-            await new JobStorageHelper(ConnectionString).DownloadJobOutputAsync(jobId, stream);
-            stream.Seek(0, SeekOrigin.Begin);
-            var output = new StreamReader(stream).ReadToEnd();
-            var deserializedOutput = JsonConvert.DeserializeObject<Dictionary<string, JToken>>(output);
-            var histogram = new Dictionary<string, double>();
-            foreach (var entry in deserializedOutput["histogram"] as JObject)
-            {
-                histogram[entry.Key] = entry.Value.ToObject<double>();
-            }
+            using var stream = new MemoryStream();
+
+            // TODO: Use JobStorageHelper.DownloadJobOutputAsync() after https://github.com/microsoft/qsharp-runtime/issues/239 is fixed.
+            // await new JobStorageHelper(ConnectionString).DownloadJobOutputAsync(jobId, stream);
+            await new JobStorageHelper(ConnectionString).StorageHelper.DownloadBlobAsync(
```

Comment:
The URL with the job output should should be read from the outputData property of CloudJob.

Reply:
Thanks, this is much more straightforward.